src/libstd/io/buffered.rs

```diff
@@ -15,9 +15,33 @@ use io::prelude::*;
 use cmp;
 use error;
 use fmt;
-use io::{self, DEFAULT_BUF_SIZE, Error, ErrorKind, SeekFrom};
+use io::{self, TrustedRead, DEFAULT_BUF_SIZE, Error, ErrorKind, SeekFrom};
 use memchr;
 
+trait MakeBuffer {
+    fn make_buffer(capacity: usize) -> Vec<u8>;
```

Comment:
Shouldn't this be `unsafe fn` given that it potentially creates an uninitialized buffer?